lib/gruf/interceptors/instrumentation/request_logging/interceptor.rb

```diff
@@ -37,6 +37,13 @@ class InvalidFormatterError < StandardError; end
         #
         class Interceptor < ::Gruf::Interceptors::ServerInterceptor
 
+          # Default mappings of codes to log levels...
+          LOG_LEVEL_MAPS = {
```

Comment:
I wonder if it would be better to store a map of gRPC status code to log level. Right now in the code, we need to first create a mapping of the level to a list of status codes, this means whenever we want to look up a status code, we need to potentially loop over the entire list of status codes. For the case of looking up an error, I need to first loop over all the info, then all the warn, then (potentially) all the errors.

It also means that if I want to override the behavior of a single status code, I don't need to re-define ALL the status codes which might be in that group.

Reply:
Correct on two fronts. I went with @splittingred 's suggestion of the hash of arrays. But this does force you to redefine them all to ensure that a code doesn't exist in multiple levels. A map per each is less elegant at first sight but a lot more efficient, both for lookup and replacement.